cli/mender-artifact/write_test.go

```diff
@@ -399,3 +401,30 @@ func TestWriteRootfsArtifactDependsAndProvides(t *testing.T) {
 	}, *updProvides)
 
 }
+
+func TestWriteRootfsImageChecksum(t *testing.T) {
+
+	// Cannot find payload file (nonexisting)
+	set := flag.NewFlagSet("provides", flag.ExitOnError)
+	set.String("file", "idonotexist", "foobar")
+	ctx := cli.NewContext(nil, set, nil)
+	err := writeRootfsImageChecksum(ctx, nil)
+	assert.Contains(t, writeRootfsImageChecksum(ctx, nil).Error(), "Failed to open the payload file")
+
+	// Checksum a dummy file
+	tf, err := ioutil.TempFile("", "TestWriteRootfsImageChecksum")
+	require.NoError(t, err)
+	_, err = tf.Write([]byte("foobar"))
+	require.NoError(t, err)
+	require.NoError(t, tf.Close())
+	typeInfo := artifact.TypeInfoV3{}
+	set = flag.NewFlagSet("file", flag.ExitOnError)
+	set.String("file", tf.Name(), "foo")
+	err = writeRootfsImageChecksum(
+		cli.NewContext(nil, set, nil), &typeInfo)
+	assert.NoError(t, err)
+	require.NotNil(t, typeInfo.ArtifactProvides)
+	_, ok := (*typeInfo.ArtifactProvides)["rootfs_image_checksum"]
+	assert.True(t, ok)
```

Comment:
This is probably something that can be inserted into an existing test, but I would also like to see one test for writing the whole artifact, and testing whether this info is there. I would be almost surprised if one of the existing tests aren't already complaining about the new artifact content. Have you run them all?

Reply:
Oh yeh yeh, it breaks them spectacularly, lots of them, this is in no way ready for a PR. When I create these draft PR's it's usually for code discussion, not review ;)